_posts/2023-04-14-h2oai.md

```diff
@@ -0,0 +1,90 @@
+| join #3 |`SELECT x.*, medium.id1 AS medium_id1, medium.id4 AS medium_id4, medium.id5 AS medium_id5, v2 FROM x LEFT JOIN medium USING (id2)`               |   Left join a large table (x) with a medium-sized table on integer type|
+| join #4 |`SELECT x.*, medium.id1 AS medium_id1, medium.id2 AS medium_id2, medium.id4 AS medium_id4, v2 FROM x JOIN medium USING (id5)`                    |   Join a large table (x) with a medium table on varchar type           |
+| join #5 |`SELECT x.*, big.id1 AS big_id1, big.id2 AS big_id2, big.id4 AS big_id4, big.id5 AS big_id5, big.id6 AS big_id6, v2 FROM x JOIN big USING (id3)` |   Join a large table (x) with a large table on integer type.           |
+
+You can find more information about the queries in the [Efficiency of Data Processing](https://jangorecki.gitlab.io/r-talks/2019-12-26_Mumbai_Efficiency-in-data-processing/Efficiency-in-data-processing.pdf) slides.
+
+### Modifications to the Benchmark & Hardware
+No modifications have been made to the queries or the data generation. Some scripts required minor modifications so that the current version of the library could be run. The hardware used is slightly different as the exact AWS offering the benchmark previously used is no longer available. Base libraries have been updated as well. GPU libraries were not tested. 
+
+
+AWS is a [m4.10xlarge](https://aws.amazon.com/ec2/instance-types/)
+
+- CPU model: Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz 
+- CPU cores: 40 
+- RAM model: Unknown 
+- Memory: 160GB 
+- NO GPU specifications 
+- R upgraded, 4.0.0 -> 4.2.2 
+- Python upgraded 3.\[6\|7\] -> 3.10 
+
+### Changes made to install scripts of other systems
+Pandas, Polars, Dask, and Clickhouse required changes to their setup/install scripts. The changes were relatively minor consisting mostly of syntax updates and data ingestion updates. Data ingestion did not affect the reporting timing results.
+
+
+## RESULTS
+
+<div style="position:relative; padding-bottom:10%; width: 100%; height:500px">
+	<iframe src="https://tmonster.github.io/h2oai-db-benchmark/"  title="h2o db benchmmark" style="width: 100%; height:100%"></iframe>
+</div>
+
+You can also look at the results [here](https://duckdb.github.io/h2oai-db-benchmark/). Hopefully you immediately notice that DuckDB ranks first in all of the queries (I mean, why else would we write a blog post about it 😜?). A major contributor to our increased performance is [parallel grouped aggregation](https://duckdb.org/2022/03/07/aggregate-hashtable.html) which was merged in March 2022. In addition, DuckDB now supports [enum types](https://duckdb.org/2021/11/26/duck-enum.html), which makes DuckDB `group by` aggregation even faster. [Improvements to the out-of-core hash join](https://github.com/duckdb/duckdb/pull/4970) were merged as well, further improving the performance of our joins.
```

Comment:
This is a bit too much/too tongue-in-cheek imo. Maybe just say like "DuckDB's timings have improved significantly since v0.2.7 released over two years ago".

The biggest improvement to the join benchmarks of h2oai is also the parallel result set materialization since the join benchmarks mostly benchmark table materialization - not sure if we want to mention that in some capacity.

Reply:
I can add it. Should I link to the PR? https://github.com/duckdb/duckdb/pull/3700

Reply:
Sure